### DISCUSSION TITLE
Pause, play, next, previous commands using headphone/audio device buttons

### DISCUSSION THREAD

**Discussion by bartfer** [jcorporation/myMPD#1010] [Ideas]
For example for a youtube playlist in the browser I can control it by using buttons on my headphone. It would be really nice if mympd could do this as well.
I don't know how it really works unfortunately. +I don't even know the terminology so feel free to update the title of this discussion. :)

(and sorry if duplicate, couldn't find similar)

**Comment by jcorporation** [comment]
As far I know the media buttons API works only for playback in the browser.

**Reply by bartfer** [reply]
Sorry but I don't really understand what you mean by this. So it is already available in mympd, or it is not possible to implement or?
Just checked the table at the bottom on this page: https://developer.mozilla.org/en-US/docs/Web/API/Media_Session_API
Couldn't mympd  use this mediasession api somehow? And the browsers that support it will have this feature.

edit:
another reference: https://web.dev/media-session/

**Reply by jcorporation** [reply]
With the media session api you can control only the local playback of the browser, not what MPD is playing.
Media session api supported is already implemented in myMPD for local playback. Both can be enabled in the settings.

**Reply by bartfer** [reply]
Oh, I'm starting to understand. So you can use media session api only when a local playback is running? Isn't there a way to somehow circumvent this? (hm, based on these it seems not really: https://stackoverflow.com/questions/71496451/web-audio-session-api-to-trigger-method-calls-in-app-rather-than-playing-media + https://stackoverflow.com/questions/58957895/chrome-mediasession-api-without-audio )
Could it work that mympd starts a "ghost"/empty local playback in order to be able to get events using media session API? So then mympd could send the commands to mpd?

My problem is that httpd is lagging/delayed and is **discouraged** by MPD authors themselves: https://github.com/MusicPlayerDaemon/MPD/discussions/1396#discussioncomment-2008113

(Would be great if we could solve this somehow even with a hack.)

edit.: I've just finished reading the mediasession-api-without-audio stackoverflow question and in the answer the user solved a similar problem with similar empty/blank audio solution.

**Reply by jcorporation** [reply]
I know this workaround with a empty/blank audio, but I do not like it.
